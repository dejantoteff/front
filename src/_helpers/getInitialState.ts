import { initialGet } from 'client-helpers'
import { darkblue3, green } from 'colors'

const randomFlag = initialGet({
  defaultValue: false,
  key: 'randomFlag',
})
const textToSpeechFlag = initialGet({
  defaultValue: true,
  key: 'textToSpeechFlag',
})
const points = initialGet({
  defaultValue: 0,
  key: 'points',
})
const fromLanguage = initialGet<Language>({
  defaultValue: 'DE',
  key: 'fromLanguage',
})
const toLanguage = initialGet<Language>({
  defaultValue: 'EN',
  key: 'toLanguage',
})

const roughData: RoughData = {
  info: {roughness: 0.7, fill: green},
  random: {roughness: 0.3, active: randomFlag},
  refresh: {roughness: 0.6, fill: darkblue3},
  send: {roughness: 0.5, fill: darkblue3},
  stepForward: {roughness: 0.5, fill: darkblue3},
  volumeDown: {roughness: 0.4, active: textToSpeechFlag},
}

export function getInitialState(): Store {

  return {
    fromLanguage,
    instructions: '',
    logged: false,
    name: '',
    points,
    randomFlag,
    ready: false,
    roughData,
    textToSpeechFlag,
    toLanguage,
    toggleLanguage: false,
  }
}
