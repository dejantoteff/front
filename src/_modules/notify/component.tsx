import * as React from 'react'
import { connect } from 'react-redux'
import { Container } from './container'
import { IN, OUT } from './constants'

interface NotifyProps{
  dispatch: any
  notifyStore: any
}

export class NotifyComponent extends React.Component<NotifyProps, {}> {

  constructor(props: NotifyProps) {
    super(props)
  }

  public render() {
    const flag = this.props.notifyStore.status === IN ||
      this.props.notifyStore.status === OUT

    return (<div>
        {flag &&
        <Container {...this.props.notifyStore}>
          <div>{this.props.notifyStore.message}</div>
        </Container>
      }
      </div>)
    }
  }

const connectComponent = ({ notifyStore }) => ({ notifyStore })

export const Notify = connect(connectComponent)(NotifyComponent)
