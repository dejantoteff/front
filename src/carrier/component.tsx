// STYLED_COMPONENTS
import { Logo } from '../navigation/styled/logo'
import {
  Container,
  LogoCell,
  Points,
  createIconCell,
} from './styled/grid'

const Info = createIconCell('info')
const Random = createIconCell('random')
const Refresh = createIconCell('refresh')
const Send = createIconCell('send')
const StepForward = createIconCell('stepforward')
const VolumeDown = createIconCell('volumedown')

// IMPORTS
import * as React from 'react'
import rough from 'roughjs'

import { connect } from 'react-redux'
import { LANGUAGE_SEPARATOR, LEARNING_MEME } from '../constants'

import { infoPath } from './icons/info'
import { randomPath } from './icons/random'
import { refreshPath } from './icons/refresh'
import { sendPath } from './icons/send'
import { stepForwardPath } from './icons/stepForward'
import { volumeDownPath } from './icons/volumeDown'

import { defaultTo } from 'rambdax'
import { LanguagesComponent } from './languages'

interface RoughDataInterface{
  [namespace: string]: {
    roughness?: number
    fill?: string
  }
}

const Paths = {
  infoPath,
  randomPath,
  refreshPath,
  sendPath,
  stepForwardPath,
  volumeDownPath
}

const RoughData: RoughDataInterface = {
  info: {roughness: 0.7, fill: 'red'},
  random: {roughness: 0.5, fill: 'teal'},
  refresh: {roughness: 0.5, fill: 'teal'},
  send: {roughness: 0.5, fill: 'teal'},
  stepForward: {roughness: 0.5, fill: 'teal'},
  volumeDown: {roughness: 0.5, fill: 'teal'},
}

function paint(){
  Object.keys(RoughData).map(
    namespace => {
      const canvasElement = rough.canvas(
        document.getElementById(`icon_${namespace.toLowerCase()}`),
      )
    
      const roughness = defaultTo(
        0.7,
        RoughData[namespace].roughness,
      )
      const fill = defaultTo(
        'green',
        RoughData[namespace].fill,
      )
      const path = Paths[`${namespace}Path`]
    
      canvasElement.path(
        path,
        { roughness, fill },
      )
    }
  )
}

/**
 * Carrier component that is shared across all components.
 * It holds navigation and icons.
 */
export class Carrier extends React.Component<Props, {}> {
  public componentDidMount(){
    paint()
  }
  public render() {
    const from = this.props.store.fromLanguage
    const to = this.props.store.toLanguage
    const name = this.props.store.name

    return (
      <Container>

        <LogoCell><Logo id='toggle-navigation' /></LogoCell>

        <Info.outer><Info.inner /></Info.outer>
        
        <Refresh.outer><Refresh.inner /></Refresh.outer>
        <Random.outer><Random.inner /></Random.outer>
        <VolumeDown.outer><VolumeDown.inner /></VolumeDown.outer>
        <Send.outer><Send.inner /></Send.outer>
        <StepForward.outer><StepForward.inner /></StepForward.outer>
        
        <Points>{this.props.store.points}</Points>
      </Container>
    )
  }
}

const connectComponent = ({ store }) => ({ store })

export const CarrierWrapped = connect(connectComponent)(Carrier)
