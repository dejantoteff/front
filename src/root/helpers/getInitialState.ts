import { initialGet } from 'client-helpers'

const randomFlag = initialGet({
  defaultValue: false,
  key: 'randomFlag',
  type: 'boolean',
})
const textToSpeechFlag = initialGet({
  defaultValue: true,
  key: 'textToSpeechFlag',
  type: 'boolean',
})
const points = initialGet({
  defaultValue: 0,
  key: 'points',
  type: 'number',
})
const fromLanguage = initialGet<Language>({
  defaultValue: 'DE',
  key: 'fromLanguage',
})
const toLanguage = initialGet<Language>({
  defaultValue: 'EN',
  key: 'toLanguage',
})

export function getInitialState(): Store {
  return {
    fromLanguage: fromLanguage,
    instructions: '',
    logged: false,
    name: '',
    points: points,
    randomFlag: randomFlag,
    ready: false,
    textToSpeechFlag: textToSpeechFlag,
    toLanguage: toLanguage,
    toggleLanguage: false,
  }
}
