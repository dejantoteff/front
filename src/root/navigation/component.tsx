import { CCell } from './styled/cells'
import { Container, Grid } from './styled/grid'

const First = CCell('nav_first')
const Second = CCell('nav_second')
const Third = CCell('nav_third')
const Fourth = CCell('nav_fourth')
const Fifth = CCell('nav_fifth')
const Sixth = CCell('nav_sixth')

import { last } from 'rambdax'
import * as React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { LEARNING_MEME } from '../../constants'


export class Navigation extends React.Component<NavigationProps, {}> {
  public render() {
    const isHome = last(window.location.href.split('/')) === ''
    const isX = this.props.store.name === LEARNING_MEME && !isHome

    return (
      <div>
        {this.props.navigationStore.active &&
          <Container>
            <Grid>
              <First>
                {
                  !isHome &&
                  <span>
                    <Link to='/learning-meme'>Learning Meme</Link>
                  </span>
                }
              </First>

              <Second>
                <span><Link to='/write-sentence'>Write Sentence</Link></span>
              </Second>

              <Third>
                <span><Link to='/choose-word'>Choose Word</Link></span>
              </Third>

              <Fourth>
                <span><Link to='/guess-word'>Guess Word</Link></span>
              </Fourth>

              <Fifth>
                <span><Link to='/user'>User</Link></span>
              </Fifth>

              <Sixth>
                {!isX && <span><Link to='/'>Home</Link></span>}
              </Sixth>
            </Grid>
          </Container>}
      </div>
    )
  }
}

const connectComponent = ({
  navigationStore,
  store,
}) => ({ navigationStore, store })

export const NavigationWrapped = connect(connectComponent)(Navigation)
