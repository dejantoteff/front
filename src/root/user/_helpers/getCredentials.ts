import { getter } from 'client-helpers'

interface Output {
  userDBName: string
  password: string
}

export function getCredentials(): Output | false {
  const userDBName = getter<string>('userDBName')
  const password = getter<string>('password')

  return userDBName && password ?
    { userDBName, password } :
    false
}
