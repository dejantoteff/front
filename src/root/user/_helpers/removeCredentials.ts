export function removeCredentials() {
  localStorage.setItem('points', '0')
  localStorage.removeItem('userDBName')
  localStorage.removeItem('password')

  window.location.reload(false)
}
