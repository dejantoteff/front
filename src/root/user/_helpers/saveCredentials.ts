import { setter } from 'client-helpers'

export function saveCredentials(userDBName: string, password: string) {
  setter('userDBName', userDBName)
  setter('password', password)
}
