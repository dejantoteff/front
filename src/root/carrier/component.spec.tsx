import { shallow } from 'enzyme'
import { identity } from 'rambdax'
import * as React from 'react'
import { getInitialState } from '../../_helpers/getInitialState'
import { Carrier } from './component'

const initialState = getInitialState()

const CurrentProps: Props = {
  dispatch: identity,
  store: initialState,
}

test('renders the message', () => {
  expect(
    shallow(<Carrier {...CurrentProps} />),
  ).toMatchSnapshot()
})
