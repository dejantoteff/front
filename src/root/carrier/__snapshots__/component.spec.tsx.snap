// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the message 1`] = `
<styled.div>
  <Styled(styled.div)>
    <styled.div
      id="toggle-navigation"
    />
  </Styled(styled.div)>
  <styled.div>
    <styled.canvas />
  </styled.div>
  <styled.div>
    <styled.canvas />
  </styled.div>
  <styled.div>
    <styled.canvas />
  </styled.div>
  <styled.div>
    <styled.canvas />
  </styled.div>
  <styled.div>
    <styled.canvas />
  </styled.div>
  <styled.div>
    <styled.canvas />
  </styled.div>
  <Styled(styled.div)
    id="points"
  >
    0
  </Styled(styled.div)>
</styled.div>
`;
